File: 0137. 只出现一次的数字 II/位运算.swift
class Solution {
    func singleNumber(_ nums: [Int]) -> Int {
        var a = 0, b = 0
        for num in nums {
            let t_a = (~a & b & num) | (a & ~b & ~num)
            let t_b = ~a & (b ^ num)
            a = t_a
            b = t_b
        }
        return b
    }
}

File: 0477. 汉明距离总和/位运算.js
// 148ms
var totalHammingDistance = function(nums) {
    let len = nums.length, result = 0
    for (let i = 0; i < 31; i += 1) {
        let one = 0
        for (let num of nums) {
            one += (num >> i & 1)
        }
        result += one * (len - one)
    }
    return result
};